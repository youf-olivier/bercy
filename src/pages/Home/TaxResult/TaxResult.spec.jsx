import { render, screen } from "@testing-library/react";
import { TaxResult } from "./TaxResult";

describe("Tax Result test suite", () => {
  it("SnapShot testing", () => {
    const { asFragment } = render(
      <TaxResult taxRate="34" taxAmount="3444" numberOfShares="2" />
    );
    expect(asFragment()).toMatchSnapshot();

    expect(screen.getAllByRole("definition")[0]).toHaveTextContent("34 %");
    expect(screen.getAllByRole("definition")[1]).toHaveTextContent("3444 â‚¬");
    expect(screen.getAllByRole("definition")[2]).toHaveTextContent("2");
  });
});
