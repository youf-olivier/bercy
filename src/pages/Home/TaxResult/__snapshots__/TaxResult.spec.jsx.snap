// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tax Result test suite SnapShot testing 1`] = `
<DocumentFragment>
  <h1
    class="af-title--content"
  >
    Résultat
  </h1>
  <div
    class="tax-result"
  >
    <dl
      class="af-restitution__listdef"
    >
      <dt
        class="af-restitution__listdef-item"
      >
        <span
          class="af-restitution__text"
        >
          Taux d'imposition
        </span>
      </dt>
      <dd
        class="af-restitution__listdef-value"
      >
        34 %
      </dd>
    </dl>
    <dl
      class="af-restitution__listdef"
    >
      <dt
        class="af-restitution__listdef-item"
      >
        <span
          class="af-restitution__text"
        >
          Montant impôt
        </span>
      </dt>
      <dd
        class="af-restitution__listdef-value"
      >
        3444 €
      </dd>
    </dl>
    <dl
      class="af-restitution__listdef"
    >
      <dt
        class="af-restitution__listdef-item"
      >
        <span
          class="af-restitution__text"
        >
          Nombre de part
        </span>
      </dt>
      <dd
        class="af-restitution__listdef-value"
      >
        2
      </dd>
    </dl>
  </div>
</DocumentFragment>
`;
